relayer/chains/evm/provider.go

```diff
@@ -177,19 +174,20 @@ func (p *Provider) WaitForResults(ctx context.Context, tx *ethTypes.Transaction)
 	ticker := time.NewTicker(DefaultPollingInterval)
 	defer ticker.Stop()
 	counter := 0
-	ctx, cancel := context.WithTimeout(ctx, DefaultMinedTimeout)
-	defer cancel()
 	for {
 		select {
 		case <-ctx.Done():
 			return nil, ctx.Err()
 		case <-ticker.C:
-			if counter >= MaximumWaitTry {
-				return nil, errors.New("Retry Limit Exceeded while waiting for results of transaction")
+			if counter >= MaximumPollTry {
+				return nil, fmt.Errorf("failed to get receipt after %d tries", counter)
 			}
 			counter++
 			txr, err := p.client.TransactionReceipt(ctx, tx.Hash())
-			if err != nil && err == ethereum.NotFound {
+			if err == nil {
```

Comment:
Can you refactor the block so the checks would be similar to normal scenario like error checks first and then other cases ?

Reply:
`if err == nil` is more frequent on mainnet, just optimization.

Reply:
Do we have any metrics to support this or how high is the impact as this is against the general recommended practice ?

Reply:
I don't have any metrics at all, this is just best practice in our case, this is early exit stretgy where you some performance gains.
Runtime will spend less time.

This is also applies to smart contract to save gas fees.

Reply:
Without metrics how can we trade off readability and best practices? The case might be different for other languages with different error-handling mechanisms, but based on plain assumption, should we move ahead this way ?